zerver/lib/event_types.py

```diff
@@ -907,6 +907,32 @@ class EventTypingStop(EventTypingStopCore):
     topic: str | None = None
 
 
+class EventTypingEditMessageStartCore(BaseEvent):
+    type: Literal["typing_edit_message"]
+    op: Literal["start"]
+    message_type: Literal["direct", "stream"]
+    sender: TypingPerson
+    message_id: int
+
+
+class EventTypingEditMessageStart(EventTypingEditMessageStartCore):
+    # TODO: fix types to avoid optional fields
+    channel_id: int | None = None
+
+
+class EventTypingEditMessageStopCore(BaseEvent):
+    type: Literal["typing_edit_message"]
+    op: Literal["stop"]
+    message_type: Literal["direct", "stream"]
+    sender: TypingPerson
+    message_id: int
+
+
+class EventTypingEditMessageStop(EventTypingEditMessageStopCore):
+    # TODO: fix types to avoid optional fields
+    channel_id: int | None = None
```

Comment:
I think having two classes, `TypingEditChannelMessageStop` and `TypingEditDirectMessageStop`, which both subclass a `EventTypingEditMessageStopCore`, would be a lot cleaner here.